### ISSUE TITLE
[Bug Report] TikTok download fails with "The read operation timed out"

### ISSUE THREAD

**Issue by bagarwa** [JunkFood02/Seal#387]
### Describe the bug

Whenever I try to download any video from TikTok, either by directly sharing from TikTok or by copy pasting the link, the download fails in Seal.

### To Reproduce

- Open TikTok app and watch any video. 
- Click share on any video and tap the ... icon at the end of apps row. 
- Android's share sheet opens. Tap Seal here. 
- Seal opens, with the TikTok video link auto-entered in its input field. 
- Video format and other options of download are selected. Tap the download button. 
- The little spinning animation runs for few seconds and then the error msg is displayed.

### Error reports & Screenshots

`ERROR: (generic] Unable to download webpage: The read operation timed out (caused by timeout(' The read operation timed out')); please report this issue on https://github.com/yt-dlp/yt-dip/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using yt-dip -U`

![Screenshot_20221204-235134](https://user-images.githubusercontent.com/4424058/205582972-5ef5f602-d0c6-4985-bb76-a79128e9d089.png)

### Device info

App version: 1.7.0-beta.2 (10610) (Also tried in 1.6.0, 1.6.1)
Device information: Android 13 (API 33)
[arm64-v8a]
Screen resolution: 2918 x 1440

Pixel 7 Pro.

### Additional context

Screenshot of Settings screen - 
![Screenshot_20221204-235211](https://user-images.githubusercontent.com/4424058/205583541-c9a3866a-3c94-4108-a75e-32c8aa81eef2.png)

**Comment by JunkFood02** [comment]
It seems this link doesn't point to the video directly, when you paste it into a browser you can get the [real URL](https://www.tiktok.com/@itsbenreid/video/7173365906746264878) of the video.

**Comment by bagarwa** [comment]
That's a little difficult to do on mobile though. First I have to copy the link from tiktok, then go to app drawer n launch Chrome, paste that link in chrome, then share that converted link from Chrome to Seal. 

Would it be possible for Seal to convert the first TikTok url to that second url internally?

**Comment by xxzozaxx** [comment]
I hope seal do it for me too.

Also I hope seal download the un-stabbed version by default

**Comment by BobbyESP** [comment]
For me, with that url, the video downloading works fine

**Comment by bagarwa** [comment]
Thank you soooo much for adding the feature I requested. Hope it can be further fine tuned though. The current implementation looks a little confusing. 

For example, after entering the url, this is how the format selection page looks like - 
[https://i.imgur.com/2rypFGT.jpg](https://i.imgur.com/2rypFGT.jpg)

Grid of 10+ tiles; with what looks like is supposed to be different resolution version of the file; each file split into 3 parts (-0, -1, -2); but I think I'm reading it wrong. Honestly I couldn't figure out what's the difference between any of those. So I went with the "suggested" one here and that was ok. 

But yeah, if possible, any better formatting of this screen would be nice.
